lib/variable/operations_common.h

```diff
@@ -28,15 +28,22 @@ SCIPP_VARIABLE_EXPORT Variable &nanmean_impl(const Variable &var, const Dim dim,
                                              const Variable &masks_sum,
                                              Variable &out);
 
-template <class T> T normalize_impl(const T &nominator, const T &denominator) {
-  // Nominator may be an int or a Eigen::Vector3d => use double
+template <class T> T normalize_impl(const T &numerator, T denominator) {
+  // Numerator may be an int or a Eigen::Vector3d => use double
   // This approach would be wrong if we supported vectors of float
   const auto type =
-      nominator.dtype() == dtype<float> ? dtype<float> : dtype<double>;
-  return nominator *
+      numerator.dtype() == dtype<float> ? dtype<float> : dtype<double>;
+  denominator.setUnit(units::one);
+  return numerator *
          reciprocal(astype(denominator, type, CopyPolicy::TryAvoid));
 }
 
+template <class T> void normalize_inplace_impl(T &numerator, T denominator) {
+  denominator.setUnit(units::one);
+  numerator *= reciprocal(
+      astype(denominator, core::dtype<double>, CopyPolicy::TryAvoid));
```

Comment:
Why do both `normalize_*_impl` functions multiply by the reciprocal instead of dividing by the denominator? That would avoid the intermediate variable.

Reply:
I think the reason was that usually the denominator is small (may just be a scalar), so it is more important to use a cheap `*` instead of `/` for every element.